workers/repositorygcworker.py

```diff
@@ -79,6 +79,11 @@ def create_gunicorn_worker():
         while True:
             time.sleep(100000)
 
+    if app.config.get("DISABLE_PUSHES", False):
+        logger.debug("Pushes to the registry are disabled; skipping startup")
+        while True:
+            time.sleep(100000)
+
```

Comment:
Wondering if these workers should be disabled in this case. If we're allowing _some_ operations, like adding or deleting tags, the GC workers should work normally no?

Couldn't we only disable the v2 endpoints concerned?

Reply:
A possible scenario where `gc` could potentially cause problems is as follows: 

1. Disable pushes on registry for maintenance purposes (for instance, one wants to upgrade Quay).
2. Create a snapshot of the database. 
3. Delete something from the database after a snapshot has been taken. GC process would start up to 30 seconds after deletion.
4. Create a snapshot of storage blobs in the backend storage while GC is taking place.
5. Restore the backed up database from your snapshot to revert changes.

The outcome here is that now we cannot guarantee consistency between what's stored in the db vs. what's stored in the storage engine. There's a possibility that the restored snapshot contains references to blobs that were deleted during the GC process which would break pulls of images. So we need to keep them in sync. The disabled pushes would be a temporary thing anyway so even if GC is turned off for that duration, it should not cause problems (the items should be GCed after read only is enabled).